halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/services/v1/DeploymentService.java

```diff
@@ -42,8 +45,26 @@
   private ValidateService validateService;
 
   @Autowired
+  private HalconfigParser halconfigParser;
+
+  @Autowired
   private PersistentStorageService storageService;
 
+  public void setDeploymentConfiguration(String deploymentName, DeploymentConfiguration deploymentConfiguration) {
+    Halconfig halconfig = halconfigParser.getHalconfig();
+    List<DeploymentConfiguration> deploymentConfigurations = halconfig.getDeploymentConfigurations();
+
+    for (int i = 0; i < deploymentConfigurations.size(); i++) {
+      DeploymentConfiguration test = deploymentConfigurations.get(i);
+      if (test.getName().equals(deploymentName)) {
+        deploymentConfigurations.set(i, deploymentConfiguration);
+        return;
```

Comment:
Might read easier to have one block to locate the index of the matching config, and then another to set the new config. As it stands now, it looks like you are setting a new config while you are still searching through the list.